### PR TITLE
[Brave News]: Update Discover section to match new design spec.

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/26462

[Feature](https://user-images.githubusercontent.com/7678024/199397004-1476a74a-bac0-432f-a0a4-6bfb3b809c05.webm)

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by petemill — CHANGES_REQUESTED** [review]
The Suggestions sections is showing when there are no suggested items. Perhaps `Carousel` can just `return null` if there are no publisher Ids...

**Comment by fallaciousreasoning** [comment]
Good call @petemill! I've updated it.

File: components/brave_new_tab_ui/components/default/braveToday/customize/CustomizeLink.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import styled from 'styled-components'
+
+const CustomizeLink = styled.button`
+  all: unset;
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 18px;
+  color: var(--interactive5);
+  cursor: pointer;
+
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/braveToday/customize/CustomizeLink.tsx:15]: nit: empty line

File: components/brave_new_tab_ui/components/default/braveToday/customize/CustomizePage.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+const Header = styled.span`
+  font-weight: 500;
+  font-size: 16px;
+  color: var(--text01);
+  flex: 5;
+  text-align: center;
+`
+
+const Spacer = styled.div`flex: 1;`
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/braveToday/customize/CustomizePage.tsx:38]: This is an interesting way to center the text. It breaks down if the word "Back" is extremely long, but that's very unlikely!

**Review by petemill — APPROVED** [review]

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.47.48 Chromium: 107.0.5304.91 (Official Build) nightly (x86_64)
-- | --
Revision | 3d5948960d62418160796d5831a4d2d7d6c90fa8-refs/branch-heads/5304@{#1097}
OS | macOS Version 13.1 (Build 22C5033e)

### Steps:
1. installed `1.47.48`
2. launched Brave
3. clicked on `Skip welcome tour`
4. opened `brave://flags`
5. set `brave://flags/#brave-news-v2` to `Enabled`
6. clicked `Relaunch`
7. opened a new-tab page
8. clicked on `Customize`
9. clicked on `Brave News`
10. clicked on `Turn on Brave News`
11. checked the `Customize Brave News` dialog

#### Confirmed the following changes:
 - `Popular` sources appeared first
 - `Suggestions` appeared second (after following a few sources, then returning to this dialog)
 - `Channels` appeared third
 - All channels were visible (no `Show more` button)
 - `Popular` & `Suggestions` appeared in a carousel, with a `View All` link taking me to a full page view
 - removed previous `Sources` section

`order of appearance` | `all channels visible` | `carousel`
-----------------------|----------------------|------------
Screenshot 2022-11-05 at 10 08 00 AM | Screenshot 2022-11-05 at 10 11 37 AM | Screenshot 2022-11-05 at 10 12 02 AM

**Pull request merged**